### PR TITLE
chore: improve Dockerfile cache handling and optimize image layers

### PR DESCRIPTION
**Pull request by Light13008**
**Description**

chore: improve Dockerfile cache handling and optimize image layers
-In this PR, i tried to solve this [Issue](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKm&tab=code)
-In this PR, i modified Dockerfile.
-here is the test file 
[Testresults.txt](https://github.com/user-attachments/files/17265812/Testresults.txt)

**Related issue(s)**

### CODE REVIEW

**Comment by Light13008** [comment]
mb, i will fix this in some time.

File: apps/generator/test/test-templates/react-template/__transpiled/test-file.md.js
Comment (Gmin2) [inline apps/generator/test/test-templates/react-template/__transpiled/test-file.md.js:1]: remove this file from PR

File: package-lock.json
Comment (Gmin2) [inline package-lock.json:1]: why have you commited the package-lock.json?

File: apps/generator/Dockerfile [hunk 1]
```diff
@@ -11,9 +11,11 @@ ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD true
 # Since 0.30.0 release Git is supported and required as a dependency
 # Since 0.14.0 release of html-template chromium is needed for pdf generation.
 # More custom packages for specific template should not be added to this dockerfile. Instead, we should come up with some extensibility solution.
-RUN apk --update add git chromium && \
-    rm -rf /var/lib/apt/lists/* && \
-    rm /var/cache/apk/*
+#is the following code correct for removing the cache after installing packages or stor it in cache mount?
```
Comment (Gmin2) [inline apps/generator/Dockerfile:14]: you are trying to fix the sonarcloud issue or ask question by this comment?

Comment (Light13008) [reply apps/generator/Dockerfile:14]: Its a solution im trying to provide.
And sorry for that Comment 😐,  pl ignore it, I will remove it

**Pull request closed without merging**